File: rust/src/solutions/s0006.rs
use proconio::input;
use std::time::Instant;

// Solution for Project Euler problem 6
// Copyright michimani All rights reserved.
//
// https://projecteuler.net/problem=6
pub fn solve() {
    let org: usize = 100;
    println!("(original: {})", org);
    input! {
      n: usize,
    }

    let start = Instant::now();

    let mut answer = 0;

    for i in 1..(n + 1) {
        for j in 1..(n + 1) {
            if i != j {
                answer = answer + (i * j)
            }
        }
    }

    println!("answer is {}", answer);

    let end = start.elapsed();
    println!(
        "\nIt took {}.{:03} seconds.",
        end.as_secs(),
        end.subsec_nanos() / 1_000_000
    );
}

File: python/solution/s0006.py
#
# Solution for Project Euler problem 6
# Copyright michimani All rights reserved.
#
# https://projecteuler.net/problem=6
#

FROM_NUM = 1
TO_NUM = 100


def solve():
    sum_of_square = 0
    sum_of_range = 0
    for num in range(FROM_NUM, TO_NUM + 1):
        sum_of_range += num
        sum_of_square += num**2

    square_of_sum = sum_of_range**2

    ans = square_of_sum - sum_of_square
    return ans


if __name__ == "__main__":
    print(solve())
